pkg/api/validation/validation.go

```diff
@@ -735,6 +743,32 @@ func validateConfigMapVolumeSource(configMapSource *api.ConfigMapVolumeSource, f
 	return allErrs
 }
 
+func validateHugePagesVolumeSource(hugePages *api.HugePagesVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// TODO(pprokop): add support for other PageSizes
+	if hugePages.PageSize != "2M" {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("pageSize"), hugePages.PageSize, "Not supported page size"))
+	}
+	maxSize, err := resource.ParseQuantity(hugePages.MaxSize)
```

Comment:
even though, some integer value might be invalid for hugepages. For example, suppose the pagesize is 2M, how many pages should that be for minSize or maxSize being 5M? 2 pages or 3 pages or other number?